### ISSUE TITLE
Error with mixing typescript and javascript imports

### ISSUE THREAD

**Issue by wgottschalk** [pahen/madge#207]
```
$ madge -V
3.4.4
```
I'm having trouble getting madge to follow `ts` imports from `js` files. For instance if I define:

```ts
// one.ts
import {two} from './two';

export function one() {
  console.log('one');
  two()
}
```

```ts
// two.js
import {three} from './three';

export function two() {
  console.log('two');
  three()
}
```

```ts
// three.ts
import {four} from './four';

export function three() {
  console.log('three');
  four()
}
```

```ts
// four.js
export function four() {
  console.log('four');
}
```

when I run madge on the source code:
`madge -j one.ts`

I get the following output:
```
{
  "one.ts": [
    "two.js"
  ],
  "two.js": []
}
```
But I'm expecting madge to be able to resolve the typescript import from a javascript file. Is there any documentation on how to do this? It seems like a common enough case.

**Comment by pahen** [comment]
https://github.com/dependents/node-dependency-tree/issues/104

**Comment by pahen** [comment]
Can you see if it works for you in Madge 3.6.0? See https://github.com/pahen/madge#using-mixed-import-syntax-in-the-same-file

**Comment by daniel-nagy** [comment]
I have a mixed codebase (JavaScript and TypeScript) and a mixed module syntax (both ES and CommonJS). TypeScript modules are not resolved from JavaScript files. I tried using a **.madgerc** with

```json
{
  "detectiveOptions": {
    "es6": {
      "mixedImports": true
    },
  "ts": {
      "mixedImports": true
    }
  }
}
```

```shell
$ npx madge -V
3.6.0
$ npx madge --image graph.svg path/to/entry.js
```

The above command only resolve JavaScript imports

**Comment by mrjoelkemp** [comment]
Just released v7.2.2 that includes some awesome lower-level work from @pastak to support this if you have the allowJs compiler option set in your tsconfig: https://github.com/dependents/node-dependency-tree/pull/122/files#diff-238d326e3d59f5c97cf05a32853368f31b662b9052647187047b7fd67b0b15d7R5.

There's a test in dependency-tree to prove that it works. 

@pahen this can be resolved now.

**Comment by pahen** [comment]
Nice job @mrjoelkemp and @pastak! 🥇 Added a note about this in Madge v3.12.0. See https://github.com/pahen/madge#mixing-typesscript-and-javascript-imports
